crates/noirc_evaluator/src/ssa_refactor/ssa_builder/mod.rs

```diff
@@ -152,44 +161,52 @@ impl FunctionBuilder {
     /// Expects the given block to be within the same function. If you want to insert
-    /// instructions into a new function, call new_function instead.
+    /// instructions into a new function, call `new_function` instead.
     pub(crate) fn switch_to_block(&mut self, block: BasicBlockId) {
         self.current_block = block;
     }
 
-    /// Returns the block currently being inserted into
+    /// Returns the block currently being modified
     pub(crate) fn current_block(&mut self) -> BasicBlockId {
         self.current_block
     }
 
-    /// Insert an allocate instruction at the end of the current block, allocating the
-    /// given amount of field elements. Returns the result of the allocate instruction,
+    /// Insert an allocate instruction at the end of the current block.
+    /// Returns the result of the allocate instruction,
     /// which is always a Reference to the allocated data.
     pub(crate) fn insert_allocate(&mut self) -> ValueId {
+        // TODO: Rust has .first on vectors which is confusing because
+        // TODO that returns Option<T> whereas this returns T and expects there
+        // TODO to be only one T.Perhaps change this to be `first_and_only`
+        // TODO or something to not conflate this with .first, but also
+        // TODO to convey the fact that its the only one; `exactly_one` ?
         self.insert_instruction(Instruction::Allocate, None).first()
     }
 
-    /// Insert a Load instruction at the end of the current block, loading from the given offset
-    /// of the given address which should point to a previous Allocate instruction. Note that
-    /// this is limited to loading a single value. Loading multiple values (such as a tuple)
-    /// will require multiple loads.
-    /// 'offset' is in units of FieldElements here. So loading the fourth FieldElement stored in
-    /// an array will have an offset of 3.
-    /// Returns the element that was loaded.
+    /// Insert a Load instruction at the end of the current block.
+    /// TODO: check previous description was not outdated.
+    /// TODO: where do we check that `address` is a Reference?
     pub(crate) fn insert_load(&mut self, address: ValueId, type_to_load: Type) -> ValueId {
         self.insert_instruction(Instruction::Load { address }, Some(vec![type_to_load])).first()
     }
 
-    /// Insert a Store instruction at the end of the current block, storing the given element
-    /// at the given address. Expects that the address points somewhere
-    /// within a previous Allocate instruction.
+    /// Insert a Store instruction at the end of the current block.
+    /// Storing the given element at the given address.
+    /// TODO: check previous description was not outdated
+    /// TODO: where do we check that `address` is a Reference?
+    /// TODO: where do we check that `value` is the correct type for `address?
+    /// TODO: Is `address` still correct terminology, given that we are using stores for mutable variables?
```

Comment:
Previous description was a bit outdated. I think we can update "points within a previous Allocate instruction` to `points at a previous Allocate instruction` or similar wording.

"Address" is still correct since mutable variables would be represented as an address with some value stored at that address. We can use a different term if you think it may be clearer though. Perhaps "allocation" since it more directly refers back to the allocate instruction?